test/Microsoft.TemplateEngine.TestHelper/PackageManager.cs

```diff
@@ -83,13 +86,11 @@ public string PackNuGetPackage(string projectPath)
             {
                 throw new ArgumentException($"{projectPath} doesn't exist", nameof(projectPath));
             }
-            lock (string.Intern(absolutePath.ToLowerInvariant()))
-            {
-                if (_installedPackages.TryGetValue(absolutePath, out string? packagePath))
-                {
-                    return packagePath;
-                }
 
+            _packMutex.WaitOne();
```

Comment:
`AbandonedMutexException` should be handled here (just logging and swollowing should be enough). Releasing below might be guarded in `finally` block as well (but handling abandoned named mutex is more important) - otherwise single failed run can cause failures in all other tests calling into this helper until the machine or container is recycled.